### ISSUE TITLE
bug: Cannot read property insertNode of undefine

### ISSUE THREAD

**Issue by sandangel** [angular/angular#19712]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

Sorry my english is not good but I try my best to describe the problem.

Navigating by back and forward to component with directive and animation cause error.

## Expected behavior

No error occur.

## Minimal reproduction of the problem with instructions

https://github.com/sandangel/angular-bug.git

yarn
yarn start

click on table link, then then graph link will show up. Click on it and the graph will show up, use back and forward button on mouse to navigate back and forward quickly between table and graph. Keep doing it several times and the bug will occur.

## Environment

<pre><code>
@angular/cli: 1.4.7
node: 8.7.0
os: darwin x64
@angular/animations: 4.4.5
@angular/cdk: 2.0.0-beta.12
@angular/common: 4.4.5
@angular/compiler: 4.4.5
@angular/core: 4.4.5
@angular/flex-layout: 2.0.0-beta.9
@angular/forms: 4.4.5
@angular/http: 4.4.5
@angular/material: 2.0.0-beta.12
@angular/platform-browser: 4.4.5
@angular/platform-browser-dynamic: 4.4.5
@angular/router: 4.4.5
@angular/cli: 1.4.7
@angular/compiler-cli: 4.4.5
@angular/language-service: 4.4.5
typescript: 2.5.3

this happens in angular 5.0.0-rc.2 too

Browser:
- [x] Chrome (desktop) version 61
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 8.7.0  
- Platform:   MacOS High Sierra, Linux x64

Others:

</code></pre>
Actually I sent a pull request to fix this problem but seem like no one noticed it so I post an issue here. I just add 

```javascript
if (this._fetchNamespace(namespaceId)) {
    this._fetchNamespace(namespaceId).insertNode(element, parent);
}
```
and the bug has gone.

https://github.com/angular/angular/pull/19662

**Comment by sandangel** [comment]
@tbosch, @chuckjaz  can you review this?

**Comment by matsko** [comment]
@sandangel I'm looking into it and I'll debug tomorrow. In the meantime can you setup the demo here? https://stackblitz.com/

**Comment by sandangel** [comment]
https://stackblitz.com/edit/angular-dso7hm here.

**Comment by sandangel** [comment]
ping @matsko

**Comment by matsko** [comment]
hey @sandangel I'm not having any luck reproducing the problem using the instructions provided. I clicked on "table", then "graph" and jumped back and forth between the pages at different speeds. No luck.

**Comment by sandangel** [comment]
![screen shot 2017-10-24 at 6 42 12](https://user-images.githubusercontent.com/22189661/31914671-8fd4b892-b886-11e7-9af6-d093edeb99e9.png)
I guess you should try it again

**Comment by sandangel** [comment]
https://gfycat.com/gifs/detail/ElasticImperfectAardwolf

I made a screen record. Please take a look

**Comment by sandangel** [comment]
![oct-24-2017 07-23-44](https://user-images.githubusercontent.com/22189661/31916171-5710d396-b88c-11e7-8106-350805a96742.gif)
Here is an animation gif

**Comment by sandangel** [comment]
Hope you can fix this before v5 release  @matsko

**Comment by sandangel** [comment]
Great. Can't wait until this commit is merged.

**Comment by matsko** [comment]
@sandangel I'm still unable to reproduce this issue with your demo. I booted it up, clicked on both links and hit back/forward many times rapidly. Still no error.

The project has a policy to include a test for each PR. If you have any other demo elsewhere that can be used to reproduce the error then it would help me get to the issue faster.

**Comment by sandangel** [comment]
@matsko You only need to click on 'Graph' link (not both) and hit back when the chart start to appear. I have updated the issue with screen record and gif file from stackblitz to prove that the error is actually occurred on my end device (iMac 2017, ubuntu 16.04). I can see it still happen at this time after several jumps back/forward by mouse button (4 ~ 10 times). Let me try posting this on gitter so others can test on their end device too.

**Comment by sandangel** [comment]
@matsko I have update the issue with new reproduce step and this time I'm sure 100% that you will get the error. If not, I'will give up.

**Comment by ghetolay** [comment]
Looks like a race condition, I got the error after several tries with chromium 62.

1. click graph link
2. reload page and while the app is bootstrapping hit back using my mouse (button is kinda freeze during reload) No need to go that fast.

Maybe some cpu throttling may make it easier to catch.

**Comment by lneninger** [comment]
It's working. I had a upper package.json that took control of node_modules selection. I removed it.

Thanks. Your question put me on the right path.

**Comment by matsko** [comment]
Hey guys I've been trying to reproduce this for weeks (clicking on the "graph" link and then hitting the back button right away) over and over again with no luck. I'm tempted to close this issue, but before I do is this still valid? Can you guys still reproduce it?

**Comment by sandangel** [comment]
@matsko I can still reproduce it. I think you miss the 'hit refresh' step. Please refer to @ghetolay comment or take a look again at my animation gif above

**Comment by ghetolay** [comment]
Yep still the same for me too as long as I use my mouse button to hit back very fast, I get the error.
Going for any ui button or even keyboard shortcut is too slow to provoke the error.

Maybe best repro would be a protractor test with a precise timing on that back action.

**Comment by matsko** [comment]
Alright I was able to reproduce it :)
